### PR TITLE
Merge window fixes

### PR DESCRIPTION
**Pull request by anandksalien**
### Fixes

- Reset confirmJson when trying to merge the documents.
- Stop the links in merge window from redirecting to avoid moving away from merge window.
- Reset collaboration when patching the document in server fails as a safety mechanism.
- Check if citation style menu is available before adding menu entries.
- Fix for margin box options menu positioning.
- Check version of the document , after participant list has arrived to avoid any issues.
- Basic validation of Access rights when user reconnects after offline.

### Features

- Links are now shown in a tooltip in the merge window.
image

### CODE REVIEW

File: fiduswriter/document/static/js/modules/editor/index.js [hunk 1]
```diff
@@ -340,6 +341,18 @@ export class Editor {
...
                         showSystemMessage(gettext('Your document was out of sync and has been reset.'))
                         break
+                    case 'access_right':
+                        if (data.access_right !== this.docInfo.access_rights) {
+                            if (sendableSteps(this.view.state) && !(WRITE_ROLES).includes(data.access_right)) {
+                                // If the user's new rights does not allow him to update document , then download a copy of the
+                                // same and ask him to re-open the document.
+                                this.handleAccessRightModification()
+                            } else {
+                                addAlert('info', gettext(`Your Access rights have been modified. You now have ${data.access_right} access to this document.`))
```
Comment (johanneswilm) [inline fiduswriter/document/static/js/modules/editor/index.js:351]: This needs to be written differently, as gettext() can only take simple strings. Therefore we need to use the `interpolate` function, like this:

```js
interpolate(
                gettext(
                    'Your Access rights have been modified. You now have %(accessRight)s access to this document.'
                ),
                {accessRights: data.access_right},
                true
            )
```

You can read more about interpolate here: https://docs.djangoproject.com/en/3.1/topics/i18n/translation/#interpolate

File: fiduswriter/document/static/js/modules/editor/index.js [hunk 2]
```diff
@@ -387,6 +400,31 @@ export class Editor {
...
+        // This function when invoked creates a copy of document in FW format and closes editor operation.
+        new ExportFidusFile(
+            this.getDoc({'use_current_view': true}),
+            this.mod.db.bibDB,
+            this.mod.db.imageDB
+        )
+        const accessRightModifiedDialog = new Dialog({
+            title: gettext('Access rights modified'),
+            id: "access_rights_modified",
+            body: gettext('Your access rights were modified while you were offline so we cannot save your work to the server and we download it to your computer instead. Please consider importing it into a new document.'),
```
Comment (johanneswilm) [inline fiduswriter/document/static/js/modules/editor/index.js:414]: How about:

Your access rights were modified while you were offline so we cannot save your work to the server any longer and **you need to** download it to your computer instead. Please consider importing it into a new document.

File: fiduswriter/document/static/js/modules/editor/index.js (hunk 1 above)
Comment (anandksalien) [reply fiduswriter/document/static/js/modules/editor/index.js]: Hi @johanneswilm , I've updated this message to use the interpolate function like you suggested.

File: fiduswriter/document/static/js/modules/editor/index.js (hunk 2 above)
Comment (anandksalien) [reply fiduswriter/document/static/js/modules/editor/index.js]: I've updated the message with your suggestion. I've also updated the message that we show when user's session expired when he came back online.

**Pull request merged**